### PR TITLE
Resolve minor bug in `rz_type_db_base_type_as_string`

### PR DESCRIPTION
**Pull request by MalhotraPulak**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [x] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**
On my quest to fix types in  Cutter I saw that `rz_type_db_base_type_as_string` is not returning the name of the member of structs.
Old output:
```
struct foo { int; int;  }
```

New output
```
struct foo { int a; int b;  }
```

Should I add some new lines too??
The output will be like
```
struct foo { 
int a; 
int b; 
}
```
Also I am thinking of adding a semi colon at the end so the output can be parsed back more easily.

### CODE REVIEW

**Review by ret2libc — COMMENTED** [review]
LGTM! @XVilka your call!

File: librz/type/base.c [hunk 1]
```diff
@@ -187,7 +187,7 @@ RZ_API RZ_OWN char *rz_type_db_base_type_as_string(const RzTypeDB *typedb, RZ_NO
 		RzTypeStructMember *memb;
 		rz_vector_foreach(&type->struct_data.members, memb) {
 			char *declaration = rz_type_identifier_declaration_as_string(typedb, memb->type, memb->name);
-			rz_strbuf_appendf(buf, "%s; ", declaration);
+			rz_strbuf_appendf(buf, "%s %s; ", declaration, memb->name);
```
Comment (XVilka) [inline librz/type/base.c:190]: Add also the fix and test for unions as well

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by MalhotraPulak** [comment]
@Xvilka Any thoughts on printing the members of the struct in a newline and putting a `;` at the end of struct and enums?

**Comment by XVilka** [comment]
I remember the newlines and `;` were a problem for nested structures. See, for example, the similar printing code at `librz/core/ctypes.c`. I recommend to add a few levels of nested structures/unions in your tests as well to make sure this doesn't get broken.

**Review by wargio — APPROVED** [review]

File: test/unit/test_type.c [hunk 2]
```diff
@@ -89,6 +89,8 @@ static bool test_types_get_base_type_struct(void) {
 	mu_assert_true(rz_type_atomic_str_eq(typedb, member->type, "int32_t"), "Incorrect type for struct member");
 	mu_assert_streq(member->name, "cow", "Incorrect name for struct member");
 
+	mu_assert_streq(rz_type_db_base_type_as_string(typedb, base), "struct kappa { int32_t bar; int32_t cow;  }", "Incorrect conversion of struct to string");
```
Comment (XVilka) [inline test/unit/test_type.c:102]: Could you please add also a line with a nested structures to check this case?

**Review by XVilka — CHANGES_REQUESTED** [review]

File: test/unit/test_type.c [hunk 3]
```diff
@@ -119,6 +135,8 @@ static bool test_types_get_base_type_union(void) {
 	mu_assert_true(rz_type_atomic_str_eq(typedb, member->type, "int32_t"), "Incorrect type for union member");
 	mu_assert_streq(member->name, "cow", "Incorrect name for union member");
 
+	mu_assert_streq(rz_type_db_base_type_as_string(typedb, base), "union kappa { int32_t bar; int32_t cow;  }", "Incorrect conversion of union to string");
+
```
Comment (XVilka) [inline test/unit/test_type.c:142]: What about the same for unions?

**Comment by MalhotraPulak** [comment]
In my last commit i meant `unions` not `enums`

**Review by XVilka — APPROVED** [review]

**Pull request merged**